src/core/blocks/core.cpp

```diff
@@ -639,6 +639,89 @@ struct PrependTo : public XpendTo {
   }
 };
 
+struct ForBlock {
```

Comment:
We might want to "inline" this block, can you move to the `.hpp` once it's finished?

Reply:
What does it change to have it in the header?

I noticed a bunch of blocks are defined there and use different macros to register themselves (the old factory ones). Are those macros necessary when defined in the header or should they also be refactored?

Reply:
So the header thing is just so that runtime.cpp or whatever actual compilation unit file (.cpp) can 100% inline it as it comes from header.
The old macros are not necessary anymore, the inlining happens in activateBlock:
See for example Const which doesn't use old macro anymore
https://github.com/fragcolor-xyz/chainblocks/blob/c87c6be56d9a0beef3ead89a207ccb5cc1e1b325/src/core/runtime.hpp#L194

This in optimized ASM will end up being a variable access, literally, 2/3 instructions

Reply:
Also seed this, this is how for should be done.
https://github.com/fragcolor-xyz/chainblocks/blob/c87c6be56d9a0beef3ead89a207ccb5cc1e1b325/src/core/runtime.hpp#L238